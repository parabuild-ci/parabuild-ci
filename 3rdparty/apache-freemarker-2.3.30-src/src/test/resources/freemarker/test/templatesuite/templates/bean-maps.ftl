[#ftl]
[#--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
--]


[#macro test hash desc]
${desc} (${hash?size}):
 [#list hash?keys?sort as key]
  ${key}[#list 0 .. (16 - key?length) as x] [/#list]: [@print value=hash[key]/]
 [/#list]
[/#macro]


[#macro print value="DEBUGME"]
  [#if value?is_number || value?is_string]
    ${value}[#t]
  [#elseif value?is_boolean]
    [#if value]true[#else]false[/#if][#t]
  [#else]
    UNKNOWN[#t]
  [/#if]
[/#macro]
[@test hash=m1 desc="properties only, shadow"/]

[@test hash=m2 desc="properties only"/]

[@test hash=m3 desc="nothing, shadow"/]

[@test hash=m4 desc="nothing"/]

[@test hash=m5 desc="all, shadow"/]

[@test hash=m6 desc="all"/]

[@test hash=m7 desc="simple map mode"/]


String concatenation:
  ${s1 + s2}
  ${s3 + s4}
  ${s1 + s3}
  ${s2 + s4}
