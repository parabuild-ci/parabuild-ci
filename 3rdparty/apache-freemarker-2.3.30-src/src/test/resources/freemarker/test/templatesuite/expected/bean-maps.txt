/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
properties only, shadow (6):
  age              : 27
  class            : class freemarker.test.templatesuite.TemplateTestCase$TestMapBean
  empty            : false
  location         : San Francisco
  luckyNumber      : 7
  name             : Christopher

properties only (6):
  age              : 27
  class            : class freemarker.test.templatesuite.TemplateTestCase$TestMapBean
  empty            : false
  location         : San Francisco
  luckyNumber      : 7
  name             : Chris

nothing, shadow (3):
  age              : 27
  location         : San Francisco
  name             : Chris

nothing (3):
  age              : 27
  location         : San Francisco
  name             : Chris

all, shadow (28):
  age              : 27
  class            : class freemarker.test.templatesuite.TemplateTestCase$TestMapBean
  clear            : UNKNOWN
  clone            : UNKNOWN
  containsKey      : UNKNOWN
  containsValue    : UNKNOWN
  empty            : false
  entrySet         : UNKNOWN
  equals           : UNKNOWN
  get              : UNKNOWN
  getClass         : UNKNOWN
  getLuckyNumber   : UNKNOWN
  getName          : UNKNOWN
  hashCode         : UNKNOWN
  isEmpty          : UNKNOWN
  keySet           : UNKNOWN
  location         : San Francisco
  luckyNumber      : 7
  name             : Christopher
  notify           : UNKNOWN
  notifyAll        : UNKNOWN
  put              : UNKNOWN
  putAll           : UNKNOWN
  remove           : UNKNOWN
  size             : UNKNOWN
  toString         : UNKNOWN
  values           : UNKNOWN
  wait             : UNKNOWN

all (28):
  age              : 27
  class            : class freemarker.test.templatesuite.TemplateTestCase$TestMapBean
  clear            : UNKNOWN
  clone            : UNKNOWN
  containsKey      : UNKNOWN
  containsValue    : UNKNOWN
  empty            : false
  entrySet         : UNKNOWN
  equals           : UNKNOWN
  get              : UNKNOWN
  getClass         : UNKNOWN
  getLuckyNumber   : UNKNOWN
  getName          : UNKNOWN
  hashCode         : UNKNOWN
  isEmpty          : UNKNOWN
  keySet           : UNKNOWN
  location         : San Francisco
  luckyNumber      : 7
  name             : Chris
  notify           : UNKNOWN
  notifyAll        : UNKNOWN
  put              : UNKNOWN
  putAll           : UNKNOWN
  remove           : UNKNOWN
  size             : UNKNOWN
  toString         : UNKNOWN
  values           : UNKNOWN
  wait             : UNKNOWN

simple map mode (3):
  age              : 27
  location         : San Francisco
  name             : Chris


String concatenation:
  helloworld
  helloworld
  hellohello
  worldworld
